src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -210,6 +210,13 @@ internal static Assembly LoadGeneratedAssembly(Type type, string defaultNamespac
 
                     return null;
                 }
+
+#if !FEATURE_SERIALIZATION_UAPAOT
+                if (!IsSerializerVersionMatch(serializer, type, defaultNamespace))
+                {
+                    return null;
```

Comment:
Could you try to put the Log method back here as what desktop did? It seems the log method in desktop is only used here.

Reply:
Why log the error here? This code is only used in our test.